File: Medium_FirstDuplicateValue.java
import java.util.*;

class Program {

  public int firstDuplicateValue(int[] array) {
		HashMap<Integer, Boolean> numbers = new HashMap<>();
		for (int i = 0; i < array.length; i++) {
			if (numbers.containsKey(array[i])) {
				return array[i];
			} else {
				numbers.put(array[i], true);
			}
		}
		return -1;
  }
}

File: Medium_FirstDuplicateValue.js
function firstDuplicateValue(array) {
	let numbers = {};
	for (let i = 0; i < array.length; i++) {
		if (numbers[array[i]]) {
			return array[i];
		} else {
			numbers[array[i]] = true;
		}
	}
	return -1;
}
